python/nammu/view/SyntaxHighlighter.py

```diff
@@ -227,7 +227,19 @@ def syntax_highlight(self, top_line=None, bottom_line=None,
                     color = self.tokencolorlu[tok.type][0]
                     styleline = self.tokencolorlu[tok.type][1]
                 if styleline:
-                    mylength = len(splittext[tok.lineno - 1])
+                    # `styleline` indicates whether we need to apply the style
+                    # to the whole line.
+                    if tok.lineno <= len(splittext):
+                        # Length of the full line.
+                        mylength = len(splittext[tok.lineno - 1])
+                    else:
+                        # For some reason, the line number stored in `tok` is
+                        # larger than the number of lines in `splittext`.  In
+                        # this case, ignore this token and issue a debug
+                        # message.  FIXME: this shouldn't happen.
+                        mylength = 0
+                        logger = self.controller.controller.logger
+                        logger.debug("Error in syntax_highlight")
```

Comment:
Maybe a bit more detail about what happened would be useful? (although as a debug message it wouldn't matter to the user)

Reply:
Sure, if I only knew what is happening :grimacing: